File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsCourseSettings.swift
import Foundation

// MARK: - Course settings models (M13.1)

struct MarkdownThemeCustom: Codable, Equatable, Hashable {
    var headingColor: String?
    var bodyColor: String?
    var linkColor: String?
    var codeBackground: String?
    var blockquoteBorder: String?
    var articleWidth: String?
    var fontFamily: String?

    static let seed = MarkdownThemeCustom(
        headingColor: "#0f172a",
        bodyColor: "#334155",
        linkColor: "#4f46e5",
        codeBackground: "#f1f5f9",
        blockquoteBorder: "#cbd5e1",
        articleWidth: "comfortable",
        fontFamily: "sans"
    )

    mutating func merge(_ other: MarkdownThemeCustom) {
        if let headingColor = other.headingColor { self.headingColor = headingColor }
        if let bodyColor = other.bodyColor { self.bodyColor = bodyColor }
        if let linkColor = other.linkColor { self.linkColor = linkColor }
        if let codeBackground = other.codeBackground { self.codeBackground = codeBackground }
        if let blockquoteBorder = other.blockquoteBorder { self.blockquoteBorder = blockquoteBorder }
        if let articleWidth = other.articleWidth { self.articleWidth = articleWidth }
        if let fontFamily = other.fontFamily { self.fontFamily = fontFamily }
    }
}

struct CourseUpdateRequest: Codable, Equatable {
    var title: String
    var description: String
    var published: Bool
    var startsAt: String?
    var endsAt: String?
    var visibleFrom: String?
    var hiddenAt: String?
    var scheduleMode: String
    var relativeEndAfter: String?
    var relativeHiddenAfter: String?
    var courseHomeLanding: String
    var courseHomeContentItemId: String?
    var courseTimezone: String?
    var gradeLevel: String?
}

struct CourseMarkdownThemePatch: Codable, Equatable {
    var preset: String
    var custom: MarkdownThemeCustom?
}

struct CourseHeroImageURLRequest: Codable {
    var imageUrl: String
}

struct CourseHeroPositionRequest: Codable {
    var objectPosition: String?
}

struct CourseGenerateImageRequest: Codable {
    var prompt: String
}

struct CourseGenerateImageResponse: Decodable {
    var imageUrl: String?
}

struct CourseFileUploadResponse: Decodable {
    var id: String
    var contentPath: String
    var mimeType: String?
    var byteSize: Int?
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsCourseSettings.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class MarkdownThemeCustom(
    val headingColor: String? = null,
    val bodyColor: String? = null,
    val linkColor: String? = null,
    val codeBackground: String? = null,
    val blockquoteBorder: String? = null,
    val articleWidth: String? = null,
    val fontFamily: String? = null,
) {
    companion object {
        fun seed() = MarkdownThemeCustom(
            headingColor = "#0f172a",
            bodyColor = "#334155",
            linkColor = "#4f46e5",
            codeBackground = "#f1f5f9",
            blockquoteBorder = "#cbd5e1",
            articleWidth = "comfortable",
            fontFamily = "sans",
        )
    }
}

@Serializable
data class CourseUpdateRequest(
    val title: String,
    val description: String,
    val published: Boolean,
    val startsAt: String? = null,
    val endsAt: String? = null,
    val visibleFrom: String? = null,
    val hiddenAt: String? = null,
    val scheduleMode: String,
    val relativeEndAfter: String? = null,
    val relativeHiddenAfter: String? = null,
    val courseHomeLanding: String,
    val courseHomeContentItemId: String? = null,
    val courseTimezone: String? = null,
    val gradeLevel: String? = null,
    val termId: String? = null,
)

@Serializable
data class CourseMarkdownThemePatch(
    val preset: String,
    val custom: MarkdownThemeCustom? = null,
)

@Serializable
data class CourseHeroImageURLRequest(val imageUrl: String)

@Serializable
data class CourseHeroPositionRequest(val objectPosition: String? = null)

@Serializable
data class CourseGenerateImageRequest(val prompt: String)

@Serializable
data class CourseGenerateImageResponse(val imageUrl: String? = null)

@Serializable
data class CourseFileUploadResponse(
    val id: String,
    val contentPath: String,
    val mimeType: String? = null,
    val byteSize: Int? = null,
)
